File: Backend/ParcelDelivery.Domain/ValueObjects/Interval.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ParcelDelivery.Domain.ValueObjects
{
    public class Interval
    {
        public Interval(decimal? min = null, decimal? max = null)
        {
            Min = min;
            Max = max;
        }

        public decimal? Min { get; set; }
        public decimal? Max { get; set; }

        public bool IsEmpty()
        {
            return (Min == null && Max == null);
        }
        public bool IsInInterval(decimal value)
        {
            if (Min != null & Max != null)
                return (value > Min && value <= Max);
            else if (Min != null)
                return value > Min;
            else
                return value <= Max;
        }
    }
}

File: Frontend/src/app/model/value-objects/interval.ts
export class Interval {
    min: number;
    max: number;
}

export function intervalAsString(interval: Interval) {
    if (interval) {
        if (interval.min && interval.max) {
            return `${interval.min} to ${interval.max}`;
        } else if (interval.min) {
            return `from ${interval.min}`;
        } else if (interval.max) {
            return `up to ${interval.max}`;
        }
    }
    return '-';
}
